### DISCUSSION TITLE
How can we build a scheduled job with dbt assets from two different databases based on dbt tags?

### DISCUSSION THREAD

**Discussion by GokaySoysaldi** [dagster-io/dagster#19938] [Q&A] [answered]
Hello Folks,

I have two different dbt assets with tag 'XXX'. One of my asset is in Sql Server other one is in IBM DB2.
I am trying to merge these two asset group in one single module and then trying to create a dbt asset schedule with build_schedule_from_dbt_selection depending on tag 'XXX'. My code look like this:

**__init__.py :**
all_assets = load_assets_from_modules([assets_SS, assets_DB2], key_prefix='ll')

daily_dbt_assets_schedule = build_schedule_from_dbt_selection(
    all_assets ,
    job_name="all_dbt_assets",
    cron_schedule="0 0 * * *",
    dbt_select="tag:XXX",
)

defs = Definitions(
    assets=all_assets,
    .
    .
    .
    ) 

Where assets_SS and assets_DB2 are two different python files containing only simple **@dbt_assets** function. (I am just loading my assets through manifest.)

Normally I have lot of active dagster-dbt jobs both on Sql Server and IBM DB2 but when I run my code with the example above I get an error like this:
**dagster._check.CheckError: Invariant failed. Description: Exactly one dbt AssetsDefinition is required** 

Am I trying to do something is not supported? How can I achive my goal?
Thank you for your time.

**Comment by rexledesma** [comment, marked as answer]
[`build_dbt_asset_selection`](https://docs.dagster.io/_apidocs/libraries/dagster-dbt#dagster_dbt.build_dbt_asset_selection) and [`build_schedule_from_dbt_selection`](https://docs.dagster.io/_apidocs/libraries/dagster-dbt#dagster_dbt.build_schedule_from_dbt_selection) are convenience APIs for creating a [`DbtAssetSelection`](https://docs.dagster.io/_apidocs/libraries/dagster-dbt#dagster_dbt.DbtManifestAssetSelection) or [`ScheduleDefinition`](https://docs.dagster.io/_apidocs/schedules-sensors#dagster.ScheduleDefinition) from a single [`@dbt_assets`](https://docs.dagster.io/_apidocs/libraries/dagster-dbt#dagster_dbt.dbt_assets) definition.

If you have multiple dbt assets definitions, then we recommend that you just construct `DbtAssetSelection` or `ScheduleDefinition` directly.

For example:

```python
from dagster import ScheduleDefinition, define_asset_job
from dagster_dbt import DbtManifestAssetSelection

manifest = ...

my_schedule = ScheduleDefinition(
    name="all_dbt_assets_daily_schedule",
    cron_schedule="0 0 * * *",
    job=define_asset_job(
        name="all_dbt_assets",
        selection=DbtManifestAssetSelection.build(
            manifest=manifest,
            select="tag:XXX",
        ),
    ),
)
```
